### PR TITLE
Add support for @Immutable on attributes

### PR DESCRIPTION
**Pull request by soldierkam**
This commit add support for @Immutable annotation on entity attributes. Benefits:
* lower memory consumption (no need to create copies for checking if entity is dirty or to put/read it from cache) - especially in cases where entity contains JSON documents
* faster flush operation

### CODE REVIEW

**Comment by vladmihalcea** [comment]
Thanks for the PR. Did you sign the [CLA](https://cla.jboss.org/)?

**Comment by soldierkam** [comment]
Signed.

File: hibernate-core/src/main/java/org/hibernate/type/descriptor/java/SerializableTypeDescriptor.java [hunk 1]
```diff
@@ -45,14 +44,16 @@ public S deepCopyNotNull(S value) {
...
-				type,
-				Serializable.class.equals( type )
-						? (MutabilityPlan<T>) SerializableMutabilityPlan.INSTANCE
-						: new SerializableMutabilityPlan<T>( type )
-		);
+		super( type, createMutabiltyPlan( type ) );
+	}
+
+	@SuppressWarnings({ "unchecked" })
+	private static <T> MutabilityPlan<T> createMutabiltyPlan(Class<T> type) {
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/type/descriptor/java/SerializableTypeDescriptor.java]: Love this idea of extending `@Immutable` to attributes and interpreting that in terms of `MutabilityPlan`.  One place you missed for that is in `org.hibernate.type.descriptor.java.JavaTypeDescriptorRegistry.FallbackJavaTypeDescriptor`.  Could you add this logic there as well?  Thanks

Comment (soldierkam) [inline hibernate-core/src/main/java/org/hibernate/type/descriptor/java/SerializableTypeDescriptor.java]: Fixed

**Comment by vladmihalcea** [comment]
I added the [HHH-10714](https://hibernate.atlassian.net/browse/HHH-10714) issue for this PR.

**Comment by vladmihalcea** [comment]
Thanks for contributing. I replaced this PR with [this one](https://github.com/hibernate/hibernate-orm/pull/1356) which adds some minor adjustments.

**Pull request closed without merging**